src/main/java/org/openstreetmap/atlas/checks/validation/intersections/SelfIntersectingPolylineCheck.java

```diff
@@ -25,15 +33,29 @@
  * {@link Edge}s and {@link Line}s. Both shape point and non-shape point intersections are flagged.
  *
  * @author mgostintsev
+ * @author dbaah
  */
 public class SelfIntersectingPolylineCheck extends BaseCheck<Long>
 {
-    private static final String INSTRUCTION_SHORT = "Self-intersecting polyline for feature {0,number,#}";
-    private static final String INSTRUCTION_LONG = INSTRUCTION_SHORT + " at {1}";
-    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(INSTRUCTION_SHORT,
-            INSTRUCTION_LONG);
+    private static final String AREA_INSTRUCTION = "Feature {0,number,#} has invalid geometry at {1}";
+    private static final String POLYLINE_BUILDING_INSTRUCTION = "Feature {0,number,#} is a incomplete "
+            + "building at {1}";
+
+    private static final String DUPLICATE_EDGE_INSTRUCTION = "Feature {0,number,#} has a duplicate "
+            + "Edge at {1}";
+    private static final String POLYLINE_INSTRUCTION = "Self-intersecting polyline for feature "
+            + "{0,number,#} at {1}";
+    // Excluded Highway tags
+    private static final Predicate<Taggable> INELIGIBLE_HIGHWAY_TAGS = object -> Validators
```

Comment:
I think it would be better if we controlled this in the configuration using the tagFilter like:
```
"tags.filter": "highway->!proposed&highway->!construction&highway->!footway&highway->!path"
```

This way if someone wanted to include those they could just modify the configuration instead of having to actually modify the code.

Reply:
Great idea, am I able to list multiple values w/ the `!` operator? : 

```
"tags.filter": "highway->!proposed,!construction,!footway,!path"
```

Reply:
The original check uses `tags.Filter` to eliminate all features not containing `highway`, `building`, or `waterway` tags. I intend to keep that initial filter, however, I'm unsure how to include 
```
highway->!construction&highway->!footway&highway->!path
``` 
into the original `highway->*|building->*` clause in order to make tag filtering fully dynamic.

Features must contain highway or building tags & if they contain highway tags, they cannot be one of the following: construction, footway, or path.

Reply:
Sorry for replying so late, basically "&"'s are grouped together. So to do what you want, you can do this: 
```highway->*&highway->!construction&highway->!footway&highway->!path|building->*```

This will kind of treat it like this:
```(highway->*&highway->!construction&highway->!footway&highway->!path)|building->*```

Note the braces should not be included, just giving you an idea of precedence. Also the specific highway tags need to be included after the wildcard highway tag.

Reply:
Awesome. Thanks for the clarification... re opening w/ this update